app/controllers/api/functions.php

```diff
@@ -817,32 +827,59 @@
         if ($async) {
             Resque::enqueue('v1-functions', 'FunctionsV1', [
                 'projectId' => $project->getId(),
+                'deploymentId' => $deployment->getId(),
+                'buildId' => $deployment->getAttribute('buildId', ''),
+                'path' => $build->getAttribute('outputPath', ''),
+                'vars' => $function->getAttribute('vars', []), 
+                'data' => $data,
+                'runtime' => $function->getAttribute('runtime', ''),
+                'timeout' => $function->getAttribute('timeout', 0),
+                'baseImage' => '',
                 'webhooks' => $project->getAttribute('webhooks', []),
+                'userId' => $user->getId(),
                 'functionId' => $function->getId(),
                 'executionId' => $execution->getId(),
                 'trigger' => 'http',
-                'data' => $data,
-                'userId' => $user->getId(),
-                'jwt' => $jwt
+                'jwt' => $jwt,
             ]);
 
             $response->setStatusCode(Response::STATUS_CODE_CREATED);
             $response->dynamic($execution, Response::MODEL_EXECUTION);
             return $response;
         }
 
+        /** Send variables */
+        // $vars = \array_merge($function->getAttribute('vars', []), [
+        //     'ENTRYPOINT_NAME' => $deployment->getAttribute('entrypoint', ''),
+        //     'APPWRITE_FUNCTION_ID' => $function->getId(),
+        //     'APPWRITE_FUNCTION_NAME' => $function->getAttribute('name', ''),
+        //     'APPWRITE_FUNCTION_DEPLOYMENT' => $deployment->getId(),
+        //     'APPWRITE_FUNCTION_TRIGGER' => 'http',
+        //     'APPWRITE_FUNCTION_RUNTIME_NAME' => $runtime['name'],
+        //     'APPWRITE_FUNCTION_RUNTIME_VERSION' => $runtime['version'],
+        //     'APPWRITE_FUNCTION_EVENT' => $event,
+        //     'APPWRITE_FUNCTION_EVENT_DATA' => $eventData,
+        //     'APPWRITE_FUNCTION_DATA' => $data,
+        //     'APPWRITE_FUNCTION_USER_ID' => $userId,
+        //     'APPWRITE_FUNCTION_JWT' => $jwt,
+        //     'APPWRITE_FUNCTION_PROJECT_ID' => $projectId
+        // ]);
+
         // Directly execute function.
         $ch = \curl_init();
```

Comment:
We could potentially abstract all this HTTP calls to the executor under: `src/Excutor/Excutor.php` until we have an SDK for it.

Reply:
done 👍